### PR TITLE
Fetching user fix

### PR DESCRIPTION
**Pull request by dderancourt**
a few bug fixes on fetching user, routes, auth and API call

### CODE REVIEW

**Review by laurentmahieu — DISMISSED** [review]

**Review by laurentmahieu — DISMISSED** [review]

File: frontend/src/components/Intervention.jsx [hunk 1]
```diff
@@ -9,25 +9,23 @@ const { apiSite } = require("../conf");
...
 
   useEffect(() => {
-    axios.get(`${apiSite}/vehicule/1/nextmaintenance`).then(({ data }) => {
-      const lvls = calcLevels(data);
-
-      setnextMaintenance(
-        lvls
-          .filter(vehicule => {
+    axios
+      .get(`${apiSite}/vehicule/1/nextmaintenance`, {
```
Comment (LoicBrassart) [inline frontend/src/components/Intervention.jsx:16]: This `1` is gonna come back and bite you in the ass

File: frontend/src/components/ListCar.jsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
-
+  const token = useSelector(state => state.user.token);
   useEffect(() => {
-    axios.get(`${apiSite}/vehicule/1/nextmaintenance`).then(({ data }) => {
-      const lvls = calcLevels(data);
-      const filtered = filterFamilies(lvls);
-      setnextMaintenance(filtered);
-    });
+    axios
+      .get(`${apiSite}/vehicule/1/nextmaintenance`, {
```
Comment (LoicBrassart) [inline frontend/src/components/ListCar.jsx:13]: So is this one

File: frontend/src/index.js [hunk 3]
```diff
@@ -5,14 +5,18 @@ import "./index.css";
 import App from "./App";
 import * as serviceWorker from "./serviceWorker";
 import { BrowserRouter } from "react-router-dom";
-
+import { createStore, compose, applyMiddleware } from "redux";
+import thunk from "redux-thunk";
```
Comment (LoicBrassart) [inline frontend/src/index.js]: Nope, this I don't get

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Don't know why there is `redux-thunk`, grumpy I'll stay.

File: frontend/src/components/Intervention.jsx (hunk 1 above)
Comment (dderancourt) [reply frontend/src/components/Intervention.jsx:16]: it's a bugfix branch, this kind of problem will be resolve in another branch

File: frontend/src/components/ListCar.jsx (hunk 2 above)
Comment (dderancourt) [reply frontend/src/components/ListCar.jsx:13]: it's a bugfix branch, this kind of problem will be resolve in another branch

**Review by laurentmahieu — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
SnE

**Pull request merged**